storage-shared-benchmarking/src/main/java/com/google/cloud/storage/benchmarking/StorageSharedBenchmarkingCli.java

```diff
@@ -108,35 +120,66 @@ private void runWorkload1() {
     StorageOptions retryStorageOptions =
         StorageOptions.newBuilder().setProjectId(project).setRetrySettings(retrySettings).build();
     Storage storageClient = retryStorageOptions.getService();
-    runW1R3(storageClient);
+    try {
+      runW1R3(storageClient);
+    } catch (Exception e) {
+      System.err.println("Failed to run workload 1: " + e.getMessage());
+    }
   }
 
   private void runWorkload4() {
     RetrySettings retrySettings = StorageOptions.getDefaultRetrySettings().toBuilder().build();
     StorageOptions retryStorageOptions =
         StorageOptions.grpc().setRetrySettings(retrySettings).setAttemptDirectPath(true).build();
     Storage storageClient = retryStorageOptions.getService();
-    runW1R3(storageClient);
+    try {
+      runW1R3(storageClient);
+    } catch (Exception e) {
+      System.err.println("Failed to run workload 4: " + e.getMessage());
+    }
   }
 
-  private void runW1R3(Storage storageClient) {
-    Path tempDir =
-        tempDirLocation != null
-            ? Paths.get(tempDirLocation)
-            : Paths.get(System.getProperty("java.io.tmpdir"));
+  private void runW1R3(Storage storageClient) throws ExecutionException, InterruptedException {
     ListeningExecutorService executorService =
         MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workers));
-    List<ApiFuture<String>> workloadRuns = new ArrayList<>();
-    Range objectSizeRange = Range.of(objectSize);
     for (int i = 0; i < samples; i++) {
+      runWarmup(storageClient);
```

Comment:
This is interesting - so if you request 5 samples, it will run the warmup for every sample?

For SSB I guess it doesn't matter, because we only request one sample (unless it were server mode, which would require other changes too and we're not sure is needed yet for Java). But in general, I would expect the application only need to warm up once in the beginning, unless there are some specifics of the Java implementation I'm not aware of?

Reply:
I missed this; good catch. This line needs to be moved out of the loop;

Reply:
Hm, it just occurred to me that on a single threaded benchmark, the connections for the later data points may be in fact less warmed than the first ones, since the warmups are done with multiple threads. But, that applies in Go, not sure for Java. The effect may also be negligible within a certain time frame. 

For consistency and because we aren't sure the effect of the warmups yet in Java, I'd say move this line out of the loop. It's easier to test with the line outside of the loop, since it's not multiplying the warmup period by the number of the sample.

For SSB workload 1, yes, every data point needs to be warmed but that's because every data point runs on a fresh client. In this case the client is used across samples so it's already warmed up. Outside of the single (or perhaps low) threaded benchmarks, spending more time warming is not necessary.